advanced/sr_gui_fingertip_visualization/src/sr_gui_fingertip_visualization/tab_layouts.py

```diff
@@ -0,0 +1,344 @@
+        self._create_connections()
+
+        self._initialize_and_start_timer()
+
+    def _initialize_data_structure(self):
+        self._data_fields = ['pressure', 'temperature']
+        self._data = dict()
+        self.finger_widget = dict()
+        self._finger_frame = dict()
+        for finger in self._fingers:
+            self._data[finger] = dict()
+            for data_field in self._data_fields:
+                self._data[finger][data_field] = 0
+
+    def initialize_subscribers(self, side):
+        self.sub = rospy.Subscriber('/{}/tactile'.format(side), ShadowPST, self._tactile_data_callback)
+
+    def stop_subscribers(self, side):
+        self.sub.unregister()
+
+    def _tactile_data_callback(self, data):
+        for i, finger in enumerate(self._fingers):
+            for data_field in self._data_fields:
+                if data_field == "pressure":
+                    self._data[finger][data_field] = data.pressure[i]
+                elif data_field == "temperature":
+                    self._data[finger][data_field] = data.temperature[i]
+
+    def _init_tactile_layout(self):
+        fingers_frame = QHBoxLayout()
+        for finger in self._fingers:
+            fingers_frame.addWidget(self.init_finger_widget(finger))
+        self.main_tab_layout.addLayout(fingers_frame)
+        rospy.logwarn("inited pst tactile layout")
+        self.main_tab_layout.addLayout(fingers_frame)
+        self.setLayout(self.main_tab_layout)
+
+    def init_finger_widget(self, finger):
+        self._finger_frame[finger] = QGroupBox(finger)
+        self._finger_frame[finger].setCheckable(True)
+        self._finger_frame[finger].setSizePolicy(1, 1)
+
+        self.finger_widget[finger] = DotUnitPST(self._finger_frame[finger])
+
+        layout = QHBoxLayout()
+        layout.addWidget(self.finger_widget[finger])
+        self._finger_frame[finger].setLayout(layout)
+
+        return self._finger_frame[finger]
+
+    def _create_connections(self):
+        self.finger_selection_show_selected_button.clicked.connect(self._button_action_show_selected_fingers)
+        self.finger_selection_show_all_button.clicked.connect(self._button_action_show_all)
+
+    def timerEvent(self):
+        for finger in self._fingers:
+            if self._finger_frame[finger].isChecked():
+                self.finger_widget[finger].update_data(self._data[finger])
+
+
+class BiotacType(Enum):
+    SP_PLUS = 0
+    SP_MINUS = 1
+    BLANK = 3
+
+
+class BiotacVisualizationTab(GenericTabLayout):
+    def __init__(self, tab_name, painter):
+        super().__init__(tab_name, painter)
+
+        self._version = "v2"  # Autodetect
+        self._electrode_count = 0
+        self._selected_fingers = []
+
+        self._initialize_data_structure()
+        self.initialize_subscriber('rh')
+
+        self._init_tactile_layout()
+        self._create_connections()
+
+        self._initialize_and_start_timer()
+
+    #  to detect if biotac is type sp+, sp-, or blank
```

Comment:
Would it be better to add this kind of comment with """ """ form and below declaring the function so that it appears in help? Or it doesn't matter now since this is a draft?